src/literal.jl

```diff
@@ -50,13 +50,15 @@ function assign_data!(lp::LiteralProto, a::Vector{<:XLAScalar})
     if eltype(a) == Bool
         lp.preds = vec(a)
     elseif eltype(a) == Int8
-        lp.u8s = vec(a)
+        lp.s8s = vec(a)
     elseif eltype(a) == Int16
         lp.u16s = collect(reinterpret(UInt8, vec(a)))
```

Comment:
I'm not sure what the `collect(reinterpret(UInt8, vec(a)))` is supposed to be doing, it doesn't make sense to me that `lp.u16s` is a `Array{UInt8,1}` instead of an `Array{UInt16,1}`.

Reply:
ProtoBuf doesn't have 16 bit primitives so it's stored as little-endian uint8s. The bug is that we're assigning to u16s not s16s on the LHS.